troposphere/static/js/components/projects/detail/resources/tableData/instance/Name.react.js

```diff
@@ -15,7 +15,7 @@ export default React.createClass({
       var instance = this.props.instance,
           name = instance.get('name').trim() || "[no instance name]";
 
-      if (!instance.id) {
+      if (!instance && !instance.get('id')) {
```

Comment:
On further inspection if `!instance && !instance.get('id')` is `True`, then we'll get an empty span. I think you meant `if (instance && !instance.get('id'))`.

We could also write this as `if (instance && isntance.isNew())`